pkg/volume/secret/secret.go

```diff
@@ -91,6 +92,10 @@ var wrappedVolumeSpec = &volume.Spec{
 }
 
 func (sv *secretVolume) SetUpAt(dir string) error {
+	if volumeutil.IsReady(sv.GetPath()) {
```

Comment:
This is going to create a file in the path that containers can access.  Is that really what you intended?  What if I create a secret named "ready"?